search-provider/eks-discovery-feed-provider.c

```diff
@@ -220,47 +244,79 @@ typedef enum {
   DISCOVERY_FEED_SET_CUSTOM_TITLE = 1 << 0
 } DiscoveryFeedCustomProps;
 
-static void
-article_card_descriptions_cb (GObject *source,
+static gboolean
+models_and_shards_for_result (EkncEngine   *engine,
+                              const gchar  *application_id,
                               GAsyncResult *result,
-                              gpointer user_data)
+                              GSList       **models,
+                              GSList       **shards,
```

Comment:
Yeah, that's a good point. Actually, I realised that we weren't cleaning up properly here since we were using `g_autoptr` on the returned list which only does `g_slist_free` and not `g_slist_free_full`.